istioctl/pkg/writer/compare/sds/util.go

```diff
@@ -0,0 +1,226 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sdscompare
+
+import (
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"time"
+
+	"istio.io/istio/istioctl/pkg/util/configdump"
+	"istio.io/istio/security/pkg/nodeagent/sds"
+	"istio.io/pkg/log"
+)
+
+// SecretItemDiff represents a secret that has been diffed between nodeagent and proxy
+type SecretItemDiff struct {
+	Agent string `json:"agent"`
+	Proxy string `json:"proxy"`
+	SecretItem
+}
+
+// SecretItem is an intermediate representation of secrets, used to provide a common
+// format between the envoy proxy secrets and node agent output which can be diffed
+type SecretItem struct {
+	Name        string `json:"resource_name"`
+	Data        string `json:"cert"`
+	Source      string `json:"source"`
+	Destination string `json:"destination"`
+	SecretMeta
+}
+
+// SecretMeta holds selected fields which can be extracted from parsed x509 cert
+type SecretMeta struct {
```

Comment:
How much of this struct is overlapped to some existing structs in pilot/pkg/security/model and things in `security/pkg/`?

Reply:
The closest match for this would probably be the [CertOptions struct](https://github.com/istio/istio/blob/f6002411cc042232d0b69e7aeac1f5585ed8eea3/security/pkg/pki/util/generate_cert.go#L38), this struct is a strict subset of that one (with a few differences around `TTL` and not having a private key).